samples/Dantooine/Dantooine.Client/Server/Startup.cs

```diff
@@ -0,0 +1,174 @@
+            var requestConfig = new ForwarderRequestConfig { ActivityTimeout = TimeSpan.FromSeconds(100) };
+
+            app.UseEndpoints(endpoints =>
+            {
+                endpoints.MapRazorPages();
+                endpoints.MapControllers();
+                endpoints.Map("/api/DantooineApi1", async httpContext =>
+                {
+                    await forwarder.SendAsync(httpContext, "https://localhost:44343/", httpClient, requestConfig, transformer);
+                }).RequireAuthorization();
+
+                endpoints.MapFallbackToPage("/_Host");
+            });
+        }
+
+        public static HeaderPolicyCollection GetSecurityHeaderPolicy(bool isDev, string idpHost)
+        {
+            var policy = new HeaderPolicyCollection()
+                .AddFrameOptionsDeny()
+                .AddXssProtectionBlock()
+                .AddContentTypeOptionsNoSniff()
+                .AddReferrerPolicyStrictOriginWhenCrossOrigin()
+                .AddCrossOriginOpenerPolicy(builder =>
+                {
+                    builder.SameOrigin();
+                })
+                .AddCrossOriginResourcePolicy(builder =>
+                {
+                    builder.SameOrigin();
+                })
+                .AddCrossOriginEmbedderPolicy(builder =>
+                {
+                    builder.RequireCorp();
+                })
+                .AddContentSecurityPolicy(builder =>
+                {
+                    builder.AddObjectSrc().None();
+                    builder.AddBlockAllMixedContent();
+                    builder.AddImgSrc().Self().From("data:");
+                    builder.AddFormAction().Self().From(idpHost);
+                    builder.AddFontSrc().Self();
+                    builder.AddStyleSrc().Self();
+                    builder.AddBaseUri().Self();
+                    builder.AddFrameAncestors().None();
+
+                    builder.AddScriptSrc()
+                        .Self()
+                        .WithHash256("v8v3RKRPmN4odZ1CWM5gw80QKPCCWMcpNeOmimNL2AA=")
+                        .UnsafeEval();
+                })
+                .RemoveServerHeader()
+                .AddPermissionsPolicy(builder =>
+                {
+                    builder.AddAccelerometer().None();
+                    builder.AddAutoplay().None();
+                    builder.AddCamera().None();
+                    builder.AddEncryptedMedia().None();
+                    builder.AddFullscreen().All();
+                    builder.AddGeolocation().None();
+                    builder.AddGyroscope().None();
+                    builder.AddMagnetometer().None();
+                    builder.AddMicrophone().None();
+                    builder.AddMidi().None();
+                    builder.AddPayment().None();
+                    builder.AddPictureInPicture().None();
+                    builder.AddSyncXHR().None();
+                    builder.AddUsb().None();
+                });
```

Comment:
```suggestion
                });
```

Reply:
Why remove this? This should be blocked if possible

Reply:
Well, it adds a lot of lines for a quite limited security gain: the sample itself doesn't use these features, so trying to block iframes from indirectly accessing these features is pointless. The only other case is if a malicious actor was able to inject custom scripts using these features via a XSS flaw. And frankly, there are way more exciting things to do with a XSS (all these features require explicit consent from the users anyway 😄)

(also, permissions policies are still experimental and browser support is fairly limited: https://developer.mozilla.org/fr/docs/Web/HTTP/Headers/Feature-Policy)

Reply:
removed